### PR TITLE
Development

### PR DESCRIPTION
**Pull request by chuanli2**
I'd like to open a pull request,  It is not ready to merge.  I still have to update documentation and make minor tweaks to the code before its ready.  

Please stand by

I've added: 
- ability to change models in get wave spectral data  and field data

- I've added get cshore data

### CODE REVIEW

**Review by SBFRF — CHANGES_REQUESTED** [review]
please update port calls (8080) add docstrings for new functions

**Comment by chuanli2** [comment]
Removed 8080s and added docstring for getCSHOREOutput. Ready to merge.

File: getDataFRF.py [hunk 1]
```diff
@@ -2426,7 +2437,7 @@ def getStwaveField(self, var, prefix, local=True, ijLoc=None):
         assert field[var].shape[0] == len(field['time']), " the indexing is wrong for pulling down bathy"
         return field
 
-    def getWaveSpecSTWAVE(self, prefix, gaugenumber, local=True):
+    def getWaveSpecSTWAVE(self, prefix, gaugenumber, local=True, model='STWAVE'):
```
Comment (SBFRF) [inline getDataFRF.py:2444]: Try to do something like this.  I think its a good way to not break things, but will move things forward as well! 

    def getWaveSpecSTWAVE(self, prefix, gaugenumber, local=True, model='STWAVE'):
        warnings.warn('Using depricated function name')
        return self.getWaveSpecSTWAVE(prefix, gaugenumber, local, model)

    def getWaveSpecModel(self, prefix, gaugenumber, local=True, model='STWAVE'):

File: getDataFRF.py [hunk 2]
```diff
@@ -1996,7 +1999,7 @@ def __init__(self, d1, d2, THREDDS='FRF'):
         self.THREDDS = THREDDS
         self.callingClass = 'getDataTestBed'
         self.FRFdataloc = u'http://134.164.129.55/thredds/dodsC/FRF/'
-        self.crunchDataLoc = u'http://134.164.129.62:8080/thredds/dodsC/CMTB/'
+        self.crunchDataLoc = u'http://134.164.129.55:/thredds/dodsC/cmtb/'
```
Comment (SBFRF) [inline getDataFRF.py]: : needs to be removed

File: getDataFRF.py [hunk 3]
```diff
@@ -2489,29 +2502,25 @@ def getWaveSpecSTWAVE(self, prefix, gaugenumber, local=True):
...
         # Making gauges flexible
         if prefix in ['CB', 'HP', 'CBHP', 'FP']:
-            model = 'STWAVE'
             urlFront = 'waveModels/%s/%s' % (model, prefix)
         elif prefix.startswith('S') and prefix[1].isdigit():  # this is static bathy
-            model = 'STWAVE'
             urlFront = 'projects/%s/CBHP/SingleBathy_%s' % (model, prefix[1:])
         elif prefix in ['CBThresh_0']:
-            model = 'STWAVE'
-            urlFront = 'projects/STWAVE/CBThresh_0'
```
Comment (SBFRF) [inline getDataFRF.py:2499]: :+1:

File: getDataFRF.py [hunk 4]
```diff
@@ -2619,3 +2630,40 @@ def getLidarWaveProf(self, removeMasked=True):
...
         return out
+
+    def getCSHOREOutput(self, prefix):
+        """retrives data from spatial data STWAVE model
+            
+        Args:
+            prefix (str): a 'key' to select which version of the simulations to pull data from
+                available value is only 'MOBILE_RESET' for now but more could be 
+                added in the future
+
```
Comment (SBFRF) [inline getDataFRF.py]: please define output dictionary (similar to above functions)

File: getDataFRF.py [hunk 5]
```diff
@@ -114,7 +116,7 @@ def __init__(self, d1, d2, THREDDS='FRF'):
         self.epochd2 = nc.date2num(self.d2, self.timeunits)
         self.THREDDS = THREDDS
         self.callingClass = 'getObs'
-        self.FRFdataloc = u'http://134.164.129.55/thredds/dodsC/FRF/'
+        self.FRFdataloc = u'http://134.164.129.55:/thredds/dodsC/FRF/'
```
Comment (SBFRF) [inline getDataFRF.py]: please remove ':'

**Review by SBFRF — COMMENTED** [review]
Here are a few things i found that would need to be fixed before an approval

File: getDataFRF.py (hunk 1 above)
Comment (SBFRF) [reply getDataFRF.py:2444]: also this would work for the STWAVE field function

Comment (chuanli2) [reply getDataFRF.py:2444]: Should be like this instead right?

return self.getWaveSpecModel(prefix, gaugenumber, local, model)

Comment (chuanli2) [reply getDataFRF.py:2444]: Made the change.

File: getDataFRF.py (hunk 2 above)
Comment (chuanli2) [reply getDataFRF.py]: Done

File: getDataFRF.py (hunk 4 above)
Comment (chuanli2) [reply getDataFRF.py]: Done

File: getDataFRF.py (hunk 5 above)
Comment (chuanli2) [reply getDataFRF.py]: Done

**Comment by chuanli2** [comment]
Made second round of changes. Ready to merge unless there are more change requests.

**Review by SBFRF — APPROVED** [review]
Looks Good!

**Pull request merged**